### ISSUE TITLE
Search Bar Backdrop in ActionBarSearchBox is not visible

### ISSUE THREAD

**Issue by jonboiser** [learningequality/kolibri#8007]
## Observed behavior

There is code for a backdrop that should be shown when the search box is open on small screens. But it doesn't look like it is ever visible. Is this just cruft?

https://github.com/learningequality/kolibri/blob/a9d1f4b579804d587b90580a0a9eeaf1a7dee955/kolibri/plugins/learn/assets/src/views/ActionBarSearchBox.vue#L29

In Chrome devtools, it looks like it's  a zero-height div in the ActionBar.

![CleanShot 2021-04-14 at 13 39 18](https://user-images.githubusercontent.com/10248067/114776187-0cdc9480-9d27-11eb-8e8c-f807b8cd8b98.png)

## Expected behavior

Backdrop is shown, or otherwise it's removed in the code.

**Comment by Priyaraj17** [comment]
Hey @jonboiser, I have checked the issue. I think it is a zero-height div. What should be the approach to solve it?

**Comment by kmrinal19** [comment]
@jonboiser @Priyaraj17 I think that adding `height: 100vh` to the following class would do the trick
https://github.com/learningequality/kolibri/blob/a9d1f4b579804d587b90580a0a9eeaf1a7dee955/kolibri/plugins/learn/assets/src/views/ActionBarSearchBox.vue#L125-L133

**Comment by Priyaraj17** [comment]
@kmrinal19 This might work. But let's wait for the reaction and approval of the mentors @jonboiser.

**Comment by jonboiser** [comment]
@kmrinal19 That approach basically, works, but might need to move the `top` by a few pixels. I'm wondering if the right thing is to just remove this, since nobody even noticed it was missing.

@jtamiace Should we just remove this backdrop? Here is what it looks like. It doesn't seem to really add anything.

![image](https://user-images.githubusercontent.com/10248067/115789593-37a79800-a37a-11eb-906d-cc550fd5e7b6.png)

**Comment by Priyaraj17** [comment]
Shall I work on this??

**Comment by Priyaraj17** [comment]
May I work on this?

**Comment by jonboiser** [comment]
@Priyaraj17 I first want to get a confirmation whether or not we want to bring back the backdrop or delete it.

**Comment by Priyaraj17** [comment]
@jonboiser okay sure. Thanks

**Comment by jtamiace** [comment]
Sorry for the late response - let's keep the backdrop in, since it focuses the user on entering keywords, and tapping the backdrop would allow them to dismiss the search input field without accidentally tapping on other interactive elements on the page

**Comment by Priyaraj17** [comment]
@jonboiser @jtamiace I have worked on this issue and solved the issue. Shall I create a pull request?

**Comment by Priyaraj17** [comment]
Infact the 100 vh height is not cover the entire page, if someone scrolls down, the backdrop won't be visible. 150 vh is working fine.

**Comment by singh-ravi-siso** [comment]
@Priyaraj17 giving a static height isn't a good practice, instead, a dynamic approach should work here.

**Comment by ravvis** [comment]
@Priyaraj17 , giving a static height isn't a good solution here, since the height can vary in the future.

The issue is with its parent element(`.scrolling-header`), which has a transform property, that is restricting the viewport to its height.

![Screenshot from 2021-05-01 23-35-12](https://user-images.githubusercontent.com/48203006/116791222-6d652300-aad6-11eb-805c-f7bc80b33a93.png)

**Comment by Priyaraj17** [comment]
Okay..Let me see what can be done here. Thanks

**Comment by Priyaraj17** [comment]
@singh-ravi-siso Are you sure this is the problem?  As things are working fine. I have tested for every small screen device

**Comment by Priyaraj17** [comment]
@jonboiser what do you think sir?

**Comment by jonboiser** [comment]
> Infact the 100 vh height is not cover the entire page, if someone scrolls down, the backdrop won't be visible. 150 vh is working fine.

I don't understand how a user can scroll when the backdrop is visible. If the backdrop is visible, you should not be able to scroll the content behind it, nor should you be able to to scroll the backdrop itself. Similarly, you should not be able to cause the scrolling header to disappear. It might be possible that the backdrop is sized too tall, which could cause the page to overflow in the vertical direction. We want the backdrop to not do that, so you might need to use a more precise height, or use something like absolute or fixed positioning. 

Basically, once the backdrop is activated, everything should be in a fixed layout and not be scrollable.

**Comment by Priyaraj17** [comment]
Hey @jonboiser, I tried to use a different approach to solve this issue but I couldn't find one, and I used a precise height. I tested for different heights and I think this will solve this issue. But if there is something else that I can do to improve it, I will work on it. I hope it works. Thanks!
